src/xcoobee/sdk/Bees.js

```diff
@@ -1,247 +1,254 @@
+   *
+   * ```js
+   * let socialBees = [];
+   *
+   * function collectSocialBees(pagingResponse) {
+   *   const { result } = pagingResponse;
+   *   const pageOfBees = result.data;
+   *   socialBees = socialBees.concat(pageOfBees);
+   *   if (pagingResponse.hasNextPage()) {
+   *     pagingResponse.getNextPage().then(collectSocialBees);
+   *   } else {
+   *     socialBees.forEach(bee => {
+   *       const { bee_system_name, description, bee_icon, ...etc } = bee;
+   *       // DO something with this data.
+   *     });
+   *   }
+   * }
+   *
+   * sdk.bees.listBees('social')
+   *   .then(collectSocialBees)
+   *   .catch(res => {
+   *     const { error } = res;
+   *     console.error(error);
+   *   });
+   * ```
+   *
+   * @async
+   * @param {string} searchText - The search text.  It is a string of keywords to
+   *  search for in the bee system name or label in the language of your account.
+   * @param {Config} [config] - If specified, the configuration to use instead of the
+   *   default.
+   *
+   * @returns {Promise<PagingResponse, ErrorResponse>} - The response.
+   * @property {number} code - The response status code.
+   * @property {Error} [error] - The response error if status is not successful.
+   * @property {string} [error.message] - The error message.
+   * @property {string} request_id - The ID of the request generated by the XcooBee
+   *   system.
+   * @property {Object} [result] - The result of the response if status is successful.
+   * @property {Bee[]} result.data - Bees for this page.
+   * @property {Object} [result.page_info] - The page information.
+   * @property {boolean} result.page_info.has_next_page - Flag indicating whether there is
+   *   another page of data to may be fetched.
+   * @property {string} result.page_info.end_cursor - The end cursor.
+   *
+   * @throws {XcooBeeError}
+   */
+  async listBees(searchText = null, config = null) {
+    this._assertValidState();
+
+    const fetchPage = async (apiCfg, params) => {
+      const { apiKey, apiSecret, apiUrlRoot } = apiCfg;
+      const { after, limit, searchText: search } = params;
+      const apiAccessToken = await this._.apiAccessTokenCache.get(apiUrlRoot, apiKey, apiSecret);
+      const beesPage = await BeesApi.bees(apiUrlRoot, apiAccessToken, search, after, limit);
+      return beesPage;
+    };
+    const apiCfg = SdkUtils.resolveApiCfg(config, this._.config);
+    const params = { searchText };
+
+    return SdkUtils.startPaging(fetchPage, apiCfg, params);
+  }
+
+  /**
+   *
+   * @async
+   * @param {Array<Object<string, Object>>} bees - A mapping of bee names to bee parameters.
+   * @param {string} bees[].key - The bee name.  A 'transfer' bee will be ignored.
+   * @param {Object} bees[].value - The bee parameters.
+   * @param {Object} options - The bee take off options.
+   * @param {Object} options.process -
+   * @param {Array<{value: any, name: string}>} [options.custom] - custom properties
```

Comment:
Any reason we can't use a JavaScript object instead of an array of objects with name and value properties? The one big difference is that the array of objects allows duplicate names? If we don't want to allow duplicate names, then a JS object is better and more standard in JavaScript.

Reply:
I did it in a same way it is in php sdk
I understood it wrong?

```
$options['custom'] = [
    [
        'name'  => 'full_name',
        'value' => 'John Doe',
    ],
    [
        'name'  => 'age',
        'value' => 29
    ],
]
```

Reply:
I know when I started the JavaScript SDK last year Bilal said to use standard JavaScript convention and style in place of standard PHP convention/style. There may be exceptions to this. I am not sure if this is one. For name/value pairs, using an `Object` is more standard in JavaScript.

Did you check what format is expected for the underlying GraphQL call for these custom properties? If GraphQL is expecting a `GraphQLList` of `GraphQLObjectType`s with `name` and `value` fields, then I would leave it as is. But if GraphQL is expecting something else, then you might want to change it based on that.

Even if GraphQL is expecing a List of Objects, we can still use a JS Object that is converted to an array of objects with name/value properties.

Reply:
Team:
I would suggest we follow JS convention in JS SDK and use object notation rather than array.
The initial design goal does not allow duplication of names for custom properties.

Reply:
@dhurlburtusa you are right, GraphQL expects a List of Objects, but we may convert JS object to required format